packages/astro/src/actions/runtime/virtual/server.ts

```diff
@@ -137,6 +137,21 @@ export function formDataToObject<T extends z.AnyZodObject>(
 		while (validator instanceof z.ZodOptional || validator instanceof z.ZodNullable) {
 			validator = validator._def.innerType;
 		}
+
+		// use zod default value when key is undefined
+   		// else continue to get correct zod class
+		if(validator instanceof z.ZodDefault) {
+			if(!formData.has(key)) {
+				obj[key] = validator._def.defaultValue();
+				continue;
+			}
+
+			while (validator instanceof z.ZodOptional || validator instanceof z.ZodNullable) {
+				validator = validator._def.innerType;
+			}
```

Comment:
I'm noticing this `while` loop is copied from the clause above. Is there a need to repeat here? Or is there a way to move the check for `default` above the initial `while` loop to catch it from there?

Reply:
Oh yeah, that could be improved.  The reason for the repeat is that someone could infinitely chain these methods together in any order. I will need to figure out a way to recursively check all instances that capture that.

I have noticed here that we're only going to be checking for these 3 instances, however, someone could chain other instances like `.nullabe()`, `.or()`, etc. which might throw errors.

I'm unsure what's the best way to handle that but in this case, I can fix this one issue with `.default()` for the time being.